### DISCUSSION TITLE
[FEATURE]  Potentially expose a facility to suppress the redraw if adding a CSS class.

### DISCUSSION THREAD

**Discussion by rstrutt** [bokeh/bokeh#11432] [General]
If you change a css class on any component, you force a redraw.  From looking back through the commits, it seems that this is intentional, and required in case the css adjusts some sizing to which the rest of the application needs to be sensitive.

However... if you're adding a css class which you *know* will not have any impact to sizing, it might be nice to be able to suppress the redraw event.  

I ran into this as I was adding a css class dynamically to more finely control the colors of a disabled drop-down on a page with many plots.  The redraw of the plots was taking multiple seconds, even though the css was not affecting anything to do with any sizing.

Minimal example (in 2.3.0dev3) below.  We can see that a redraw is triggered by the presence of inner_width ModelChanged events in the websocket view, but the values are unchanged as the css class that we add/remove does absolutely nothing;

```
from bokeh.layouts import column
from bokeh.models import Button, MultiSelect, Select
from bokeh.plotting import curdoc, figure

# Get a reference to the document we will be creating
doc = curdoc()

# Create simple plot
# prepare some data
x = [1, 2, 3, 4, 5]
y = [6, 7, 2, 4, 5]

# create a new plot with a title and axis labels
plot = figure(title="simple line example", x_axis_label='x', y_axis_label='y')

# add a line renderer with legend and line thickness
plot.line(x, y, legend_label="Temp.", line_width=2)

# Create a column
col = column()

# Add a button with a callback that just sets a css class on the col
button = Button(label='Press Here')

# Add a python callback on the Button to disable the MultiSelect/Select.
def button_handler():
    if 'some_funky_css_class' in col.css_classes:
        col.css_classes = []
    else:
        col.css_classes = ['some_funky_css_class']

# Add callback
button.on_click(button_handler)

# Add plot and col to the column
col.children = [plot, button]

# Put the column into the document
doc.add_root(col)
```

**Comment by IuryPiva** [comment]
Hello @rstrutt, thanks for submitting this feature proposal and sorry this did not got a deeper look at yet. I will tag @bokeh/dev for thoughts on this and move it into a discussion for now.
